`General`: Add exam mode and feature overview docs for exam checklist and grading keys (#4967)
ls1intum/Artemis: docs/user/exams/instructors_guide.rst

.. note::
        You should delete test runs before the actual exam conduction takes place.

1.8 Exam Checklist
^^^^^^^^^^^^^^^^^^
- After you create an exam, the exam checklist appears at the top of the exam's detail page.
- The exam checklist helps you oversee and ensure every step of the exam is executed correctly.
- You can track the progress of the steps mentioned in this document and spot missed steps easily.
- Each row of the checklist includes the name of the task, description and short summary where it is applicable and the page column which navigates the instructors to the relevant action.
- Going through each task from the start until the current task and checking the description column contains no warnings or errors can help instructors conduct the exam smoothly.

.. figure:: instructor/exam_checklist.png
   :alt: Exam Checklist
   :align: center

   In Progress Exam Checklist

> - The exam list page displays a more concise overview of the exam steps in the exam status column.

Author: ``Exam Checklist`` page is the other page which is more detailed so I think it is better to avoid that term here, I committed the other change.

.. note::
        You should delete test runs before the actual exam conduction takes place.

1.8 Exam Checklist
^^^^^^^^^^^^^^^^^^
- After you create an exam, the exam checklist appears at the top of the exam's detail page.
- The exam checklist helps you oversee and ensure every step of the exam is executed correctly.
- You can track the progress of the steps mentioned in this document and spot missed steps easily.
- Each row of the checklist includes the name of the task, description and short summary where it is applicable and the page column which navigates the instructors to the relevant action.
- Going through each task from the start until the current task and checking the description column contains no warnings or errors can help instructors conduct the exam smoothly.

.. figure:: instructor/exam_checklist.png
   :alt: Exam Checklist
   :align: center

   In Progress Exam Checklist

1.9 Grading Key
^^^^^^^^^^^^^^^
- Optionally, you can create a grading key for your exam by clicking |grade_key| at the top of the exam's detail page.

> - Defining a grading key allows the exam score to be converted to a grade automatically by Artemis, students are then able to see their own grades after their submissions are assessed.

Reviewer: ```suggestion
- Defining a grading key allows the exam score to be converted to a grade automatically by Artemis, students are then able to see their own grades after the specified `Release Date of Results`.
```

> - Grading key also enhances the generated statistics so that the instructor is able to view grade distributions.

Reviewer 2: ```suggestion
- Using a grading key also enhances the generated statistics so that the instructor is able to view grade distributions.
```

- For an easy out-of-the-box configuration, you can click |generate_default_grade_key| and then click ``Save``.

> - By default, grades are defined as percentages of the total obtainable score. You can also have their point equivalent if you specify ``Maximum number of points for exam``.

Reviewer 3: ```suggestion
- By default, grades are defined as percentages of the total obtainable score. You can also display their point equivalent if you specify ``Maximum number of points for exam``.
```

Sounds a little nicer IMO.

> - If you would like to have custom grade steps, you can use the |add_grade_step| button and modify the grade step intervals.

Reviewer 3: ```suggestion
- If you would like to define custom grade steps, you can use the |add_grade_step| button and modify the grade step intervals.
```

.. note::

>     Keep an eye out for the warnings at the bottom of the page to ensure the grading key is valid.

Reviewer 2: ```suggestion
    Keep an eye out for the warnings at the bottom of the page to ensure that the grading key is valid.
```

- ``Inclusivity`` field allows you to decide which grade should be assigned if the student's score is exactly equal to a boundary value between two grades.

> - There are two grade types you can use: ``Grade`` and ``Bonus``: ``Grade`` type allows you to set a final grade for the exam with custom grade step names, ``Bonus`` type allows you to assign bonus points to each grade step so they can contribute to the grade of another course or exam.

Reviewer 2: ```suggestion
- There are two grade types you can use: ``Grade`` and ``Bonus``. The ``Grade`` type allows you to set a final grade for the exam with custom grade step names, while the ``Bonus`` type allows you to assign bonus points to each grade step so they can contribute to the grade of another course or exam.
```
